### PR TITLE
Add bcrypt check and fix to kit registration

### PR DESCRIPTION
**Pull request by squirrelo**
Does what it says on the tin.

### CODE REVIEW

**Comment by wasade** [comment]
Looks good. Just to check, did you test locally?

**Comment by squirrelo** [comment]
don't know how honestly.

**Comment by wasade** [comment]
Manually add a kit to the handouts table

**Comment by squirrelo** [comment]
Just did that and it fails. The `ag_update_kit_password` function only updates the ag_kit table, not ag_handout_kits. This seems to be an issue. Not sure how we want to proceed.

File: amgut/handlers/auth_handlers.py [hunk 1]
```diff
@@ -38,6 +39,12 @@ def post(self):
         printresults = ag_data.checkPrintResults(skid)
         if printresults is None:
             printresults = 'n'
+
+        # make sure kit has bcrypt encrypted password, and fix if not
+        if not bcrypt.identify(kitinfo['password']):
```
Comment (adamrp) [inline amgut/handlers/auth_handlers.py]: Under what circumstances does this function evaluate to `False`? The docstring on the function (`passlib.hash.bcrypt.identify`) is not helpful, and I couldn't find documentation [on their page](http://pythonhosted.org//passlib/lib/passlib.hash.bcrypt.html?highlight=bcrypt#passlib.hash.bcrypt).

Comment (wasade) [inline amgut/handlers/auth_handlers.py]: @beaugunderson, can you comment when you have a moment please?

Comment (beaugunderson) [inline amgut/handlers/auth_handlers.py]: `bcrypt.identify` returns False if the argument is not a bcrypt hash; the docs are [here](http://pythonhosted.org//passlib/password_hash_api.html?highlight=identify#passlib.ifc.PasswordHash.identify) (it's a member of all of the different hash modules so they just documented it in one place)

Comment (adamrp) [inline amgut/handlers/auth_handlers.py]: Aha! Thanks @beaugunderson

**Comment by squirrelo** [comment]
This now logs the user out if you can't register the kit, which was probably what was causing the error. User was logged in but kit errored on registration so there was a weird logged in but no kit associated limbo state.

One major issue now is that, after bcrypt encrypting the password and registering the kit, I can't login to the kit. It just has me try and register it again.

**Comment by adamrp** [comment]
Do we know why the kit errored on registration? Figuring that out would prevent the liminal state, if I understand correctly.

**Comment by squirrelo** [comment]
No, because that render error was covering up the other error. Hopefully with this fix we can see the original error.

**Comment by squirrelo** [comment]
Ok, found the error and fixed it. Should be good for testing on someone elses computer for verification.

**Comment by wasade** [comment]
Thanks @squirrelo. Sorry to drag this out a bit further, but we think it makes more sense to also force the handout kit passwords to be encrypted. Have a separate PR going in (#379) that takes care of that. This makes the identity check redundant

File: amgut/templates/no_auth_help_request.html [hunk 2]
```diff
@@ -0,0 +1,40 @@
+{% extends no_auth_sitebase.html %}
```
Comment (wasade) [inline amgut/templates/no_auth_help_request.html]: Can this template be reached? Not seeing it in any render calls

Comment (squirrelo) [inline amgut/templates/no_auth_help_request.html]: It's the base template for the site when the user is logged out, so nothing will call it directly.

**Comment by squirrelo** [comment]
Might be good to have both, just to catch stragglers and have redundancies built in. Taking the [Chaos Monkey](http://blog.codinghorror.com/working-with-the-chaos-monkey/) approach.

**Comment by wasade** [comment]
Chaos Monkey randomly breaks things, doesn't do excessive checks... not
necessarily opposed to keeping it but it shouldn't be necessary.

**Pull request merged**